### PR TITLE
Status indicators

### PR DESCRIPTION
**Pull request by emcenrue**

### CODE REVIEW

File: canon-react.js [hunk 1]
```diff
@@ -0,0 +1,196 @@
+!function(e){if("object"==typeof exports)module.exports=e();else if("function"==typeof define&&define.amd)define(e);else{var n;"undefined"!=typeof window?n=window:"undefined"!=typeof global?n=global:"undefined"!=typeof self&&(n=self),n.canonReact=e()}}(function(){var define,module,exports;return (function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);throw new Error("Cannot find module '"+o+"'")}var f=n[o]={exports:{}};t[o][0].call(f.exports,function(e){var n=t[o][1][e];return s(n?n:e)},f,f.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(_dereq_,module,exports){
```
Comment (adampickeral) [inline canon-react.js]: don't commit this file

File: canon-react.min.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+/*! canon-react 2015-06-16 */
```
Comment (adampickeral) [inline canon-react.min.js]: don't commit this file

File: demo/bundle.js [hunk 3]
```diff
@@ -0,0 +1,482 @@
+(function e(t,n,r){function s(o,u){if(!n[o]){if(!t[o]){var a=typeof require=="function"&&require;if(!u&&a)return a(o,!0);if(i)return i(o,!0);throw new Error("Cannot find module '"+o+"'")}var f=n[o]={exports:{}};t[o][0].call(f.exports,function(e){var n=t[o][1][e];return s(n?n:e)},f,f.exports,e,t,n,r)}return n[o].exports}var i=typeof require=="function"&&require;for(var o=0;o<r.length;o++)s(r[o]);return s})({1:[function(require,module,exports){
```
Comment (adampickeral) [inline demo/bundle.js]: don't commit this file

Comment (adampickeral) [inline demo/bundle.js]: add this one to .gitignore

File: .gitignore [hunk 4]
```diff
@@ -3,3 +3,6 @@ transpiled/*
 test_bundle.js
 test-built/*
 test-results.xml
+canon-react.js
```
Comment (adampickeral) [inline .gitignore]: i don't think we want to ignore this one or canon-react.min.js yet. @SethNute is working on a pre-publish script for the npm module, and ignoring these files could break that

Comment (SethNute) [inline .gitignore]: I think ignoring these will be fine. The two install cases: Cloning git rep and running <code>npm install</code> from inside the directory with <code>package.json</code> will end up running <code>npm run prepublish</code> which, finally, will run <code>npm run build</code> and generate these files. Second case, running <code>npm install canon-react</code> is going to just include these in the <code>/dist</code> directory

Comment (adampickeral) [inline .gitignore]: cool, then continue ignoring ;)

File: src/StatusIndicator.jsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
+var StatusIndicator = React.createClass({
+  propTypes: {
+    status: React.PropTypes.string,
```
Comment (adampickeral) [inline src/StatusIndicator.jsx]: let's restrict what's allowed for this prop type. here's an example https://github.com/rackerlabs/canon-react/pull/27/files#diff-90112dca5b39a1ad52fc84dc1d0d8cc9R6

File: src/StatusIndicator.jsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
+var StatusIndicator = React.createClass({
```
Comment (adampickeral) [inline src/StatusIndicator.jsx]: add `var React = require('react');` after rebasing with master

**Comment by adampickeral** [comment]
looks good to me. rebase with master to resolve conflicts

**Pull request merged**